templates/model/goldentest/v1beta3/goldentest.go

```diff
@@ -54,6 +54,14 @@ type Test struct {
 	BuiltinVars []*VarValue `yaml:"builtin_vars,omitempty"`
 }
 
+// TestCase represents a test.yaml describing test configs.
```

Comment:
It's not immediately obvious to me why we need both Test and TestCase

Reply:
If we don't have `TestCase` here and use the `Test`, as the the inputs and built in vars of `Test` can be empty, yaml marshal will emit a string as `{}`. Therefore I use a `TestCase` here with the api version and kind that won't be empty. Another approach would be use the `Test` but I need to check whether it is a empty struct or not and then use yaml marshal, I feel it is kind of less clean code.

Reply:
I sent you https://github.com/abcxyz/abc/pull/390 as an alternative approach to addressing this, let me know what you think